concent_api/core/models.py

```diff
@@ -494,70 +494,53 @@ def clean(self) -> None:
             raise ValidationError({
                 'computation_deadline': "TaskToCompute deadline mismatch"
             })
+
         # Validation for every nested message which is stored in Control database
         # Every nested message must be the same as message stored separately.
-        if deserialized_report_computed_task.task_to_compute != deserialized_task_to_compute:
-            raise ValidationError({
-                'report_computed_task': 'Nested ReportComputedTask with TaskToCompute message '
-                                        'must be the same as both messages stored separately'
-            })
-        if self.ack_report_computed_task is not None:
-            if (
-                deserialize_message(self.ack_report_computed_task.data).report_computed_task !=
-                deserialized_report_computed_task
-            ):
-                raise ValidationError({
-                    'ack_report_computed_task': 'Nested ReportComputedTask with TaskToCompute message '
-                                                'must be the same as both messages stored separately'
-                })
-        if self.reject_report_computed_task is not None:
-            deserialized_reject_report_computed_task = deserialize_message(self.reject_report_computed_task.data)
-            if deserialized_reject_report_computed_task.reason == message.tasks.RejectReportComputedTask.REASON.GotMessageCannotComputeTask.name:  # pylint: disable=no-member
-                if deserialized_reject_report_computed_task.cannot_compute_task.task_to_compute != deserialized_task_to_compute:
+        def validate_consistence_of_nested_messages(message_1: message.Message, message_2: Optional[StoredMessage]=None) -> None:
```

Comment:
Also, it is a bit complex - maybe there is a way to simplify it?

Reply:
>does it have to be an inner function?

I think, that it does not have to be inner function, but why not if this function is used only to validate message which wants to be stored to database?

>Also, it is a bit complex - maybe there is a way to simplify it?

Hmm, any example please, because I thought that we can'y simplify it more :)

Reply:
Ok, at least please change naming: let
`message_1` -> `message_from_db`
`message_2` -> `nested_message`